src/index.ts

```diff
@@ -219,12 +223,15 @@ async function runUpgrades(options: Options, timeout?: NodeJS.Timeout): Promise<
               .relative(path.resolve(pkgOptions.cwd), pkgFile!)
               // convert Windows path to *nix path for consistency
               .replace(/\\/g, '/')
-          : pkgFile!]: await runLocal(pkgOptions, pkgData, pkgFile),
+          : pkgFile!]: runResult,
       }
     }, Promise.resolve({} as Index<PackageFile> | PackageFile))
     if (options.json) {
       printJson(options, analysis)
     }
+    if (totalUpgradedCount > 0 && options.errorLevel === 2) {
```

Comment:
I think we can just check if the `analysis` is empty. Then we don't have to alter the return type or add other variables. 

See the `someUpgraded` variable in the `npmInstall` function.

Reply:
Ok perfect changed it - it is a much smaller change now. I wasn't too sure about the structure of the response